modules/redis/main.tf

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "random" {
+  version = "~> 2.0"
+}
+
+resource "random_id" "id" {
+  byte_length = 4
+}
+
+locals {
+  runid = random_id.id.hex
+}
+
+resource "google_compute_address" "redis" {
+  project      = var.project_id
+  region       = var.region[0]
+  subnetwork   = var.subnetwork
+  name         = "redis-${var.cluster_name}-${local.runid}-0"
+  address_type = "INTERNAL"
+}
+
+data "template_file" "config" {
+  template = file("${path.module}/redis.sh.tpl")
+  vars = {
+    master = google_compute_address.redis.address
+    pass   = var.pass
+  }
+}
+
+resource "google_storage_bucket_object" "config" {
+  bucket  = var.bucket_name
+  name    = "${var.cluster_name}/redis.sh"
+  content = data.template_file.config.rendered
+}
+
+resource "google_compute_instance_template" "redis" {
```

Comment:
Could we consider updating this to use https://github.com/terraform-google-modules/terraform-google-vm/tree/master/modules/instance_template and https://github.com/terraform-google-modules/terraform-google-vm/tree/master/modules/mig?

Reply:
I'll do this in a secondary PR

Reply:
SGTM